src/lib/eina/eina_thread_win32.c

```diff
@@ -0,0 +1,188 @@
+/* EINA - EFL data type library
+ * Copyright (C) 2020 Carlos Ré Signor
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdlib.h>
+#include "eina_config.h"
+#include "eina_lock.h"
+#include "eina_sched.h"
+#include "eina_cpu.h"
+#include "eina_thread.h"
+/* undefs EINA_ARG_NONULL() so NULL checks are not compiled out! */
+#include "eina_safety_checks.h"
+
+#include "eina_debug_private.h"
+
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#include <Windows.h>
+#include <errno.h>
+#include <string.h>
+
+typedef struct _Eina_win32_thread_func
+{
+   void *data;
+   void *(*func)(void *data);
+
+} Eina_win32_thread_func;
+
+typedef struct _Eina_win32_thread_attr
+{
+   LPSECURITY_ATTRIBUTES lpThreadAttributes;
+   SIZE_T dwStackSize;
+   DWORD dwCreationFlags;
+} Eina_win32_thread_attr;
+
+inline void *
+_eina_thread_join(Eina_Thread t)
+{
+
+   int ret = (int)WaitForSingleObject(t, INFINITE);
+
+   if (ret != 0)
+      return ret;
+   return NULL;
+}
+
+DWORD WINAPI _eina_thread_func(void *params)
+{
+   return (DWORD)((Eina_win32_thread_func *)params)->func((void *)((Eina_win32_thread_func *)params)->data);
+}
+
+void _eina_thread_set_priority(Eina_Thread_Priority prio, Eina_Thread *t)
+{
+
+   int nPriority;
+
+   switch (prio)
+   {
+   case EINA_THREAD_URGENT:
+      nPriority = THREAD_PRIORITY_HIGHEST;
+      break;
+   case EINA_THREAD_NORMAL:
+      nPriority = THREAD_PRIORITY_NORMAL;
+      break;
+   case EINA_THREAD_BACKGROUND:
+      nPriority = THREAD_PRIORITY_BELOW_NORMAL;
+      break;
+   case EINA_THREAD_IDLE:
+      nPriority = THREAD_PRIORITY_IDLE;
+      break;
+   }
```

Comment:
You don't have a default case. I think normal priority should be an ok default case.

Reply:
I left this question open in the previous review, because I didn't know what the default value was and suggested to return false to indicate an error. I'll keep normal priority

Reply:
I think false isn't a good idea....Normal priority as a default would be better.